src/main/java/com/google/firebase/auth/User.java

```diff
@@ -0,0 +1,394 @@
+   *
+   * @return an email address string or null.
+   */
+  public String getEmail() {
+    return email;
+  }
+
+  /**
+   * Returns whether the email address of this user has been verified.
+   *
+   * @return true if the email has been verified, and false otherwise.
+   */
+  public boolean isEmailVerified() {
+    return emailVerified;
+  }
+
+  /**
+   * Returns the display name of this user.
+   *
+   * @return a display name string or null.
+   */
+  public String getDisplayName() {
+    return displayName;
+  }
+
+  /**
+   * Returns the photo URL of this user.
+   *
+   * @return a URL string or null.
+   */
+  public String getPhotoUrl() {
+    return photoUrl;
+  }
+
+  /**
+   * Returns whether this user account is disabled.
+   *
+   * @return true if the user account is disabled, and false otherwise.
+   */
+  public boolean isDisabled() {
+    return disabled;
+  }
+
+  /**
+   * Returns the identity providers associated with this user.
+   *
+   * @return an array of {@link ProviderUserInfo} instances, which may be empty.
+   */
+  public ProviderUserInfo[] getProviderData() {
+    return providers;
+  }
+
+  /**
+   * Returns additional metadata associated with this user.
+   *
+   * @return a non-null UserMetadata instance.
+   */
+  public UserMetadata getUserMetadata() {
+    return this.userMetadata;
+  }
+
+  /**
+   * Returns a new {@link User.Updater} instance, which can be used to update the attributes
+   * of this user.
+   *
+   * @return a non-null User.Updater instance.
+   */
+  public Updater updater() {
+    return new Updater(uid);
+  }
+
+  /**
+   * Returns a new {@link User.Updater} instance, which can be used to update the attributes
+   * of the user identified by the specified user ID. This method allows updating attributes of
+   * a user account, without first having to call {@link FirebaseAuth#getUser(String)}.
+   *
+   * @param uid a non-null, non-empty user ID string.
+   * @return a non-null User.Updater instance.
+   * @throws IllegalArgumentException If the user ID is null or empty.
+   */
+  public static Updater updater(String uid) {
+    return new Updater(uid);
+  }
+
+  /**
+   * Returns a new {@link User.Builder} instance, which can be used to create a new user. The
+   * returned builder should be passed to {@link FirebaseAuth#createUser(Builder)} to register
+   * the user information persistently.
+   *
+   * @return a non-null User.Builder instance.
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  private static void checkEmail(String email) {
+    checkArgument(!Strings.isNullOrEmpty(email), "email cannot be null or empty");
+    checkArgument(email.matches("^[^@]+@[^@]+$"));
+  }
+
+  private static void checkPassword(String password) {
+    checkArgument(!Strings.isNullOrEmpty(password), "password cannot be null or empty");
+    checkArgument(password.length() >= 6, "password must be at least 6 characters long");
```

Comment:
FWIW, it seems kind of strange to have this in the SDK; I'd be inclined torely on the backend to enforce any applicable password restrictions...  and I worry that the backend might have other requirements which could manifest slightly differently which seems undesirable.